### PR TITLE
Enhance REST responses for even better error diagnostics

### PR DESCRIPTION
**Pull request by mlwilkerson**
This PR represents a Whole Nother Level ™ of defensive programming to survive the wild and chaotic environment that is (sometimes) WordPress.

It also includes some additional refinements to the development environment, motivated by a desire to be able to run the Site Health checks in local, to make sure we know what a user would/should see when running the same diagnostics.

### CODE REVIEW

File: admin/src/util/reportRequestError.js [hunk 1]
```diff
@@ -9,12 +9,16 @@ const REST_NO_ROUTE_ERROR = __( 'Oh no! Your web browser could not reach your Wo
 const REST_COOKIE_INVALID_NONCE_ERROR = __( 'It looks like your web browser session expired. Try logging out and log back in to WordPress admin.', 'font-awesome' )
 const OK_ERROR_PREAMBLE = __( 'The last request was successful, but it also returned the following error(s), which might be helpful for troubleshooting.', 'font-awesome' )
 const ONE_OF_MANY_ERRORS_GROUP_LABEL = __( 'Error', 'font-awesome' )
+const FALSE_POSITIVE_MESSAGE = __( 'The last request contained errors, though your WordPress server reported it as a success. This usually means there\'s a problem with your theme or one of your other plugins emitting output that is causing problems.', 'font-awesome' )
```
Comment (mlwilkerson) [inline admin/src/util/reportRequestError.js]: @frrrances The next few error message are new copy in this PR. Would you like to revise them? None of these would end up in UI, only in the JavaScript console when errors are reported.

File: admin/src/store/actions.js [hunk 2]
```diff
@@ -16,6 +18,84 @@ export const CONFLICT_DETECTION_SCANNER_DURATION_MIN = 10
 // (which would just be exactly "now").
 const CONFLICT_DETECTION_SCANNER_DEACTIVATION_DELTA_MS = 1
 
+const COULD_NOT_SAVE_CHANGES_MESSAGE = __( 'Couldn\'t save those changes', 'font-awesome' )
+const COULD_NOT_CHECK_PREFERENCES_MESSAGE = __( 'Couldn\'t check preferences', 'font-awesome' )
```
Comment (mlwilkerson) [inline admin/src/store/actions.js]: @frrrances this is also a new error message. Though I'm not sure whether we were showing an error message in this case previously

File: admin/src/util/reportRequestError.js (hunk 1 above)
Comment (frrrances) [reply admin/src/util/reportRequestError.js]: These seem good to me 👍

**Comment by mlwilkerson** [comment]
merging this in separately

**Pull request closed without merging**